src/App.tsx

```diff
@@ -15,48 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  byAlphabet = 'Sort alphabetically',
+  byLength = 'Sort by length',
+}
+
+function getReadyGoods(
+  goods: string[],
+  sortField: string,
+  reverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.byAlphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.byLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverse
+    ? preparedGoods.reverse()
+    : preparedGoods;
+}
+
+function reset(
+  setSortField: React.Dispatch<React.SetStateAction<string>>,
+  setReverse: React.Dispatch<React.SetStateAction<boolean>>,
+) {
+  setSortField('');
+  setReverse(false);
+}
+
+export const App : React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
```

Comment:
reverse is sounds as a functionName but it is a boolean property, so I suggest you to rename it to the` [isReversed, setIsReversed]`